File: programmers/pg_lv3_입국심사.js
function solution(n, times) {
  let result = Infinity;
  times.sort((a, b) => a - b);

  let start = 0;
  let end = times[times.length - 1] * n;

  while (start <= end) {
    const mid = Math.floor((start + end) / 2);
    let count = 0;

    for (const person of times) {
      count += Math.floor(mid / person);
    }

    if (count >= n) end = mid - 1;
    else start = mid + 1;
  }
  return start;
}

File: python/programmers/pg_lv3_입국심사.py
def solution(n, times):
    start = 0
    end = max(times) * n

    while start <= end:
        mid = (start + end) // 2
        count = 0
        
        for i in times:
            count += (mid // i)
            
        if count < n: start = mid + 1
        else: end = mid - 1
    
    return start